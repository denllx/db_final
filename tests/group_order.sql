CREATE DATABASE OOP;
USE OOP;
CREATE TABLE oop_info(stu_id INT NOT NULL, PRIMARY KEY(stu_id), stu_name CHAR,grade INT);
INSERT INTO oop_info(stu_id, stu_name,grade) VALUES (2018011243, "a",100);
INSERT INTO oop_info(stu_id, stu_name,grade) VALUES (2018011043, "a",90);
INSERT INTO oop_info(stu_id, stu_name,grade) VALUES (2018010243, "a",92);
INSERT INTO oop_info(stu_id, stu_name,grade) VALUES (2018001344, "b",93);
INSERT INTO oop_info(stu_id, stu_name,grade) VALUES (2018001394, "b",99);
INSERT INTO oop_info(stu_id, stu_name,grade) VALUES (2018011445, "c",98);
INSERT INTO oop_info(stu_id, stu_name) VALUES (2018011545, "c");
SELECT stu_name, COUNT(*) from oop_info GROUP BY stu_name ORDER BY COUNT(*);
SELECT stu_name, COUNT(*) from oop_info GROUP BY stu_name ORDER BY MAX(grade);
SELECT stu_name, COUNT(*) from oop_info GROUP BY stu_name ORDER BY MIN(grade);
SELECT stu_name, COUNT(*) from oop_info GROUP BY stu_name ORDER BY COUNT(grade);
DROP DATABASE OOP;
