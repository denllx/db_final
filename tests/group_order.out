stu_name	count(*)
a	3
b	2
c	2
stu_name	count(*)
a	3
b	2
c	2
stu_name	count(*)
c	2
b	2
a	3
stu_name	count(*)
a	3
b	2
c	2
